### DISCUSSION TITLE
It would be really awesome if the front matter parser was a little smarter, a single newline at the top of the file breaks it

### DISCUSSION THREAD

**Discussion by factoidforrest** [facebook/docusaurus#5765] [Ideas]
It made a real headache for us when our license tooling wanted to add license headers at the top of all our markdown files.  We figured out a workaround where we put the license below the front matter, but even still, it seems really brittle that a single white space at the top of the file breaks the front-matter parser.

**Comment by Josh-Cena** [comment]
The front matter is commanded to be at the very top: https://jekyllrb.com/docs/front-matter/ (didn't find any formal spec about this, but all descriptions of front matter highlights this)

It's the same case for JS hashbangs, and we used to have troubles with that as well.

Maybe we could remove comments & trim before parsing front matter

**Comment by factoidforrest** [comment]
That sounds like a good compromise to me.  Removing comments and white
space would still make sure that it's at the top of the file, without
worrying about things that shouldn't matter like a single carriage return.

**Reply by slorber** [reply]
If you want something non-std, what about using or building a remark plugin?

https://www.npmjs.com/package/remark-remove-comments

Not sure it would work though, I think remark is applied after reading the frontmatter

**Reply by slorber** [reply]
PR welcome, fix should likely be done here:

https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-utils/src/markdownParser.ts#L79

Note the `gray-matter` seems to work with frontmatter comments, why not use that instead?

https://github.com/jonschlinkert/gray-matter/blob/8a22958e0afd4b2e09c705becec1c35e76c4f0ee/test/matter-empty.js#L31
